src/odemis/util/test/linalg_test.py

```diff
@@ -127,6 +128,36 @@ def test_are_collinear(self):
         non_linear_points = numpy.array([[1, 2, 12], [-1, -2, -4], [3, 4, 22]])
         self.assertFalse(are_collinear(non_linear_points[0], non_linear_points[1], non_linear_points[2]))
 
+    def test_generate_triangulation_points(self):
+        """
+        Test the number of focus points when given area changes.
+        """
+        max_dis = 450e-06
+        length_1_start = 0
+        length_1_end = 1.77e-03
+        length_2_start = 0
+        length_2_end = 1.77e-03
+
+        # given area is 100% of maximum area (1.77mm x 1.77mm)
+        given_area_coords = (length_1_start, length_2_start, length_1_end, length_2_end)
+        focus_points = generate_triangulation_points(max_dis, given_area_coords)
+        self.assertEqual(len(focus_points), 16)
+        self.assertEqual(len(focus_points[0]), 2)  # (x,y)
+
+        # given area is 90 % maximum area
+        red_length = 1.77e-03*numpy.sqrt(0.9)
```

Comment:
I don't know why it's called "red". It sounds like a colour, but I don't get why you'd use a colour.
I'd suggest to just explicitly reduce one (or 2 dimensions) to 0 -> 3.01e-6, and indicate as a comment that it should always round up to 4x4.

BTW, I'd recommend to *not* use square area in general in the tests. Because if your code somehow inverses x and y, or has a copy-paste error where x is used twice, you'll not notice it. => Use an area with 4x5 points for instance.

Reply:
I used it as a short for reduced length that became red_length. I will explicitly mention the values and explain in the comments for better transparency.